File: java-lambda-proxy/src/main/java/com/serverless/repository/BookRepository.java
package com.serverless.repository;

import com.serverless.model.Book;

import java.util.Arrays;
import java.util.List;
import java.util.Optional;

public class BookRepository {

    private static final List<Book> books = Arrays.asList(
            new Book("1", "The Shining", 1, 1977, 12000),
            new Book("2", "Black House", 1, 2001, 8700),
            new Book("3", "Da Vinci Code", 2, 2003, 130000),
            new Book("4", "Angles & Demons", 2, 2000, 98000),
            new Book("5", "Deception Point", 2, 2001, 56000),
            new Book("6", "Les Fourmis", 3, 1991, 46800),
            new Book("7", "Le Jour des fourmis", 3, 1992, 48300),
            new Book("8", "La Révolution des fourmis", 3, 1996, 53000)
    );

    private static volatile BookRepository bookRepository;

    private BookRepository() {
    }

    public static BookRepository instance() {
        if (bookRepository == null) {
            synchronized (BookRepository.class) {
                if (bookRepository == null) {
                    bookRepository = new BookRepository();
                }
            }
        }
        return bookRepository;
    }

    public List<Book> getBooks() {
        return books;
    }

    public Optional<Book> getBook(String bookId) {
        return books.stream()
                .filter(book -> book.getId().equals(bookId))
                .findFirst();
    }

}

File: docker-and-lambda/app/repositories/BookRepository.ts
const books = [
  {
    id: 1,
    title: "The Shining",
    author_id: 1,
    year: 1977,
    likes: 12000
  },
  {
    id: 2,
    title: "Black House",
    author_id: 1,
    year: 2001,
    likes: 8700
  },
  {
    id: 3,
    title: "Da Vinci Code",
    author_id: 2,
    year: 2003,
    likes: 130000,
  },
  {
    id: 4,
    title: "Angels & Demons",
    author_id: 2,
    year: 2000,
    likes: 98000,
  },
  {
    id: 5,
    title: "Deception Point",
    author_id: 2,
    year: 2001,
    likes: 56000,
  },
  {
    id: 6,
    title: "Les Fourmis",
    author_id: 3,
    year: 1991,
    likes: 46800,
  },
  {
    id: 7,
    title: "Le Jour des fourmis",
    author_id: 3,
    year: 1992,
    likes: 48300,
  },
  {
    id: 8,
    title: "La Révolution des fourmis",
    author_id: 3,
    year: 1996,
    likes: 53000,
  }
];

export async function list () {
  return books;
};

export async function get(bookId) {
  return books.filter(elt => elt.id == bookId)[0];
};